bentoml/_internal/bento/build_config.py

```diff
@@ -403,22 +530,61 @@ class BentoBuildConfig:
     labels: t.Optional[t.Dict[str, t.Any]] = None
     include: t.Optional[t.List[str]] = None
     exclude: t.Optional[t.List[str]] = None
-    docker: t.Optional[DockerOptions] = attr.field(
-        default=None,
-        converter=_dict_arg_converter(DockerOptions),
+    docker: DockerOptions = attr.field(
+        factory=DockerOptions,
+        converter=dict_options_converter(DockerOptions),
     )
-    python: t.Optional[PythonOptions] = attr.field(
-        default=None,
-        converter=_dict_arg_converter(PythonOptions),
+    python: PythonOptions = attr.field(
+        factory=PythonOptions,
+        converter=dict_options_converter(PythonOptions),
     )
-    conda: t.Optional[CondaOptions] = attr.field(
-        default=None,
-        converter=_dict_arg_converter(CondaOptions),
+    conda: CondaOptions = attr.field(
+        factory=CondaOptions,
+        converter=dict_options_converter(CondaOptions),
     )
 
+    def __attrs_post_init__(self) -> None:
+        use_conda = any(
+            val is not None for _, val in bentoml_cattr.unstructure(self.conda).items()
+        )
+        use_cuda = self.docker.cuda_version is not None
```

Comment:
add a user warning about conda options ignored when CUDA is specified?

Reply:
done